microsoft/testsuites/vm_extensions/azureperformancediagnostics.py

```diff
@@ -0,0 +1,196 @@
+import string
+from datetime import datetime, timezone
+from typing import Any
+
+from assertpy import assert_that
+
+from lisa import (
+    Logger,
+    Node,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    simple_requirement,
+)
+from lisa.operating_system import Redhat
+from lisa.sut_orchestrator.azure.common import (
+    check_or_create_storage_account,
+    delete_storage_account,
+    get_or_create_storage_container,
+    get_storage_credential,
+)
+from lisa.sut_orchestrator.azure.features import AzureExtension
+from lisa.sut_orchestrator.azure.platform_ import AzurePlatform
+from lisa.testsuite import TestResult
+from lisa.util import SkippedException, generate_random_chars
+
+
+@TestSuiteMetadata(
+    area="vm_extension",
+    category="functional",
+    description="Tests for the Azure Performance Diagnostics VM Extension",
+)
+class AzurePerformanceDiagnostics(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node = kwargs["node"]
+        if not self._is_supported_linux_distro(node):
+            raise SkippedException(
+                f"{str(node.os.information.full_version)} is not supported."
+            )
+
+    @TestCaseMetadata(
+        description="""
+        Installs and runs the Azure Performance Diagnostics VM Extension.
+        Verifies a report was created and uploaded to the storage account.
+        Deletes the VM Extension.
+        """,
+        priority=1,
+        requirement=simple_requirement(
+            supported_features=[AzureExtension],
+        ),
+    )
+    def verify_azure_performance_diagnostics(
+        self, log: Logger, node: Node, result: TestResult
+    ) -> None:
+        environment = result.environment
+        assert environment, "fail to get environment from testresult"
+        platform = environment.platform
+        assert isinstance(platform, AzurePlatform)
+
+        # Create storage account and get credentials
+        random_str = generate_random_chars(string.ascii_lowercase + string.digits, 10)
+        storage_account_name = f"lisasc{random_str}"
+        information = environment.get_information()
+        resource_group_name = information["resource_group_name"]
```

Comment:
Don't get resource_group_name from information. it should be gotten from AzureContext of a node. Check other extension cases.